packages/trpc/server/routers/viewer/teams/publish.schema.ts

```diff
@@ -1,7 +1,11 @@
 import { z } from "zod";
 
-export const ZPublishInputSchema = z.object({
+// Define type first to use with z.ZodType annotation
+// This prevents full Zod generic tree from being emitted in .d.ts files
+export type TPublishInputSchema = {
+  teamId: number;
+};
+
+export const ZPublishInputSchema: z.ZodType<TPublishInputSchema> = z.object({
   teamId: z.coerce.number(),
```

Comment:
This is converting string|number to number.  
  
So, this too needs 3-generic pattern